nautobot/extras/forms/mixins.py

```diff
@@ -394,6 +401,37 @@ def clean(self):
           verify that the requested "source" object does not have an existing RelationshipAssociation to
           a different destination object.
         """
+
+        required_relationships = Relationship.objects.get_required_for_model(self.instance)
+        for side, relations in required_relationships.items():
+            for relation in relations:
+
+                # Not enforcing required symmetric relationships
+                if relation.symmetric:
+                    continue
+
+                # Skip self-referencing
+                if ContentType.objects.get_for_model(self.instance) == getattr(
+                    relation, f"{relation.required_side}_type"
+                ):
+                    continue
```

Comment:
should we be using `skip_required()` here instead?

Reply:
Well spotted!